astronomer/providers/amazon/aws/hooks/redshift_data.py

```diff
@@ -64,13 +67,17 @@ def get_conn_params(self) -> Dict[str, Union[str, int]]:
 
         return conn_params
 
-    def execute_query(self, sql: Union[str, Iterable[str]], params: Optional[Dict]):
+    def execute_query(
+        self, sql: Optional[Union[Dict[Any, Any], Iterable[Any]]], params: Optional[Dict[Any, Any]]
```

Comment:
In [RedshiftSQLOperator](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/redshift_sql.py#L56) the `sql: Optional[Union[Dict[Any, Any], Iterable[Any]]]` is Optional in constructor. If we remove it as optional, it will throw error as its inherited from Parent `RedshiftSQLOperator`.

Reply:
I don't see it as `None`.

https://github.com/apache/airflow/blob/c7286e53064d717c97807f7ccd6cad515f88fe52/airflow/providers/amazon/aws/operators/redshift_sql.py#L56

Reply:
aah looks like I fixed it: https://github.com/apache/airflow/pull/21885 🤦

Reply:
Following changes should take care of it:

- https://github.com/astronomer/astronomer-providers/pull/103/files#r821725255
- https://github.com/astronomer/astronomer-providers/pull/103/files#r821724962

Reply:
Since we now use `cast` we should be good to remove `Optional`